src/shared/watchdog.c

```
@@ -86,6 +88,35 @@ static int watchdog_set_timeout(void) {
         return 0;
 }
 
+static int watchdog_get_pretimeout(void) {
+        int sec = 0;
+
+        assert(watchdog_fd > 0);
+
+        if (ioctl(watchdog_fd, WDIOC_GETPRETIMEOUT, &sec) < 0)
+                return log_warning_errno(errno, "Failed to get pretimeout value, ignoring: %m");
+
+        watchdog_pretimeout = sec * USEC_PER_SEC;
+
+        return 0;
+}
+
+static int watchdog_set_pretimeout(void) {
+        int sec, r;
+
+        assert(watchdog_fd >= 0);
+        assert(watchdog_pretimeout != USEC_INFINITY);
+
+	sec = saturated_usec_to_sec(watchdog_timeout);
+
+        r = ioctl(watchdog_fd, WDIOC_SETPRETIMEOUT, &sec) < 0;
+
+        /* The set ioctl does not return the actual value set so get it now. */
+        watchdog_get_pretimeout();
```

if you ignore error return codes knowingly, please cast the function to (void).

(Static checkers care)